core/src/main/java/org/opensearch/sql/ast/expression/HighlightFunction.java

```diff
@@ -6,21 +6,29 @@
 package org.opensearch.sql.ast.expression;
 
 import java.util.List;
-import lombok.AllArgsConstructor;
+import java.util.Map;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.Setter;
 import lombok.ToString;
 import org.opensearch.sql.ast.AbstractNodeVisitor;
 
 /**
  * Expression node of Highlight function.
  */
-@AllArgsConstructor
 @EqualsAndHashCode(callSuper = false)
 @Getter
 @ToString
 public class HighlightFunction extends UnresolvedExpression {
   private final UnresolvedExpression highlightField;
+  private final Map<String, Literal> arguments;
+  @Setter
+  private String name;
+
+  public HighlightFunction(UnresolvedExpression highlightField, Map<String, Literal> arguments) {
```

Comment:
Does `@AllArgsConstructor` not generate the same constructor?

Reply:
Name is not set in the constructor. I failed to parse the exact string from `AstExpressionBuilder` when returning the `HighlightFunction` instance. String is tokenized and white space is ignored.